File: cpp/n455_DeliverCookie.cpp
// https://leetcode-cn.com/problems/assign-cookies/
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        int i = 0, j = 0, count = 0;
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        while(i < g.size() && j < s.size()){
            if (g[i] <= s[j]) {
                count += 1;
                i += 1;
                j += 1;
            }
            else j += 1;
        }
        return count;
    }
};

int main(){
    return 0;
}

File: python/n455_DeliverCookie.py
# https://leetcode-cn.com/problems/assign-cookies/

class Solution:
    def findContentChildren(self, g, s) -> int:
        cook = len(s)
        count = 0
        if cook == 0: return 0
        g.sort()
        s.sort()
        print(g, s)
        i = j = 0
        while j < cook and i < len(g):
            if g[i] <= s[j]:
                count += 1
                i += 1
                j += 1
            elif g[i] > s[j]:
                j += 1
        return count

        
if __name__ == "__main__":
    g = [10, 9, 8, 7]
    s = [5, 6, 7, 8]
    solu = Solution()
    print(solu.findContentChildren(g, s))
    